docs/concurrency-adoption-guidelines.md

```diff
@@ -0,0 +1,179 @@
+
+```swift 
+#if swift(>=5.5)
+func yourAsyncFunctionConvertedToAFuture(on eventLoop: EventLoop) 
+    -> EventLoopFuture<Result> {
+    let promise = context.eventLoop.makePromise(of: Out.self)
+    promise.completeWithTask {
+        try await yourMethod(yourInputs)
+    }
+    return promise.futureResult
+}
+#endif
+```
+
+Further helpers exist for `EventLoopGroup`, `Channel`, `ChannelOutboundInvoker` and `ChannelPipeline`.
+
+### Sendable Checking
+
+> [SE-0302][SE-0302] introduced the Sendable protocol, which is used to indicate which types have values that can safely be copied across actors or, more generally, into any context where a copy of the value might be used concurrently with the original. Applied uniformly to all Swift code, Sendable checking eliminates a large class of data races caused by shared mutable state.
+>
+> -- from [Staging in Sendable checking][sendable-staging], which outlines the Sendable adoption plan for Swift 6.
+
+In the future we will see fully checked Swift concurrency. The language features to support this are the `Sendable` protocol and the `@Sendable` keyword for closures. Since sendable checking will break existing Swift code, a new major Swift version is required.
+
+To ease the transition to fully checked Swift code, it is possible to annotate your APIs with the Sendable protocol today.
+
+You can start adopting Sendable and getting appropriate warnings in Swift 5.5 already by passing the `-warn-concurrency` flag, you can do so in SwiftPM for the entire project like so:
+
+```
+swift build -Xswiftc -Xfrontend -Xswiftc -warn-concurrency
+```
+
+
+#### Sendable checking today
+
+Sendable checking is currently disabled in Swift 5.5(.0) because it was causing a number of tricky situations for which we lacked the tools to resolve.
+
+Most of these issues have been resolved on today’s `main` branch of the compiler, and are expected to land in the next Swift 5.5 releases. It may be worthwhile waiting with adoption until the next version(s) after 5.5.0.
+
+For example, one of such capabilities is the ability for tuples of `Sendable` types to conform to `Sendable` as well. We recommend holding off adoption of `Sendable` until this patch lands in Swift 5.5 (which should be relatively soon). With this change, the difference between Swift 5.5 with `-warn-concurrency` enabled and Swift 6 mode should be very small, and manageable on a case by case basis.
+
+#### Backwards compatibility of declarations and “checked” Swift Concurrency
+
+Adopting Swift Concurrency will progressively cause more warnings, and eventually compile time errors in Swift 6 when sendability checks are violated, marking potentially unsafe code.
+
+It may be difficult for a library to maintain a “pre Swift 6” version along with a Swift 6 version that fully embraces the new concurrency checks. For example, it may be necessary to mark generic types as Sendable, like so:
```

Comment:
```suggestion
It may be difficult for a library to maintain a version that is compatible with versions prior to Swift 6 while also fully embracing the new concurrency checks. For example, it may be necessary to mark generic types as `Sendable`, like so:
```